### ISSUE TITLE
Push official images to Quay, in addition to Docker Hub

### ISSUE THREAD

**Issue by aditya-konarde** [grafana/grafana#27677]
**What would you like to be added**:
Hello everyone,

Not sure if this is a 'feature' request, please feel free to relabel :)

I would like to propose that we also push the official grafana/grafana container image to quay.io

Please see the 'Why is this needed' section for a brief explanation on why.

I'm happy to set up the quay repository and send a PR to the CI job that builds these images, once we have a general agreement from the maintainers that this is a good idea.

**Why is this needed**:

In a [recent blog](https://www.docker.com/blog/scaling-docker-to-serve-millions-more-developers-network-egress/), Docker hub announced rate limits on container pulls, in addition to many other changes to how image retention works. 

Since Grafana is one of the most widely used tools along with Prometheus, I think these rate limits would impact the production infrastructure at multiple organizations. The Prometheus images are already available on Quay, which is why it makes sense to also have the grafana/grafana image also published there. We currently mirror the image for our internal use-cases, but it would be much more awesome if we could get the official images pushed.

Please let me know what you think about this :)

**Comment by hugohaggmark** [comment]
@aditya-konarde is this related to https://github.com/grafana/grafana/issues/27380?

**Comment by aditya-konarde** [comment]
Hello @hugohaggmark ! Thanks for following up on this! I think #27380 related to the same root cause, but the tickets have different goals. 

From what I understand, #27380 deals with archiving of old image tags that currently exist on docker hub somewhere (to another registry?) so that they don't go away

This ticket (#27677) is about modifying the build pipelines to ensure all *new* tags are pushed to Quay. We can, ofcourse merge these two issues and make sure old tags also exist on Quay. That would be a good end goal, I think :)

**Comment by hugohaggmark** [comment]
@aditya-konarde I don't think we'll move to another registry, the original issue is to make sure we keep our registry by all means. With that said I'm not sure we'll have an issue with old images moving forward, but still needs to be solved thoug :)

**Comment by torkelo** [comment]
we have a paid docker hub org now so those rate limits & retention limits do not apply, but sitll worh considering quay if it's better than docker hub

**Comment by anishasthana** [comment]
We're running into image-pull back-offs on Dockerhub, and I imagine a lot of other users will be running into the same issues soon enough. Have you folks had any thoughts on pushing to both repositories instead?

**Comment by fchiorascu** [comment]
@torkelo kindly could you consider this feature as something very usefully for the end users approach/accessibility unblocking the current situation. It will be appreciated, I've searched trough quay.io repositories for grafana and prometheus but only prometheus is present there.

> Nowadays the approach of Docker Hub with new rate limiting policies make a little bit hard to use the Grafana repository where another alternative is quay.io container engine repository.

**Comment by torkelo** [comment]
@fchiorascu any reasons why you prefer quay.io over docker hub? the rate-limiting should not affect Grafana images

**Comment by fchiorascu** [comment]
@torkelo yes you are perfectly right with that statement but there is a case where if you try to deploy using the "grafana" images from docker hub inside an openshift template template for example, you will need to test first until is GA and there will be retries in dev/int environments for example where that limit will be reached and you will be stuck.

Failed to pull image "grafana/grafana": rpc error: code = Unknown desc = toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit

**Comment by anishasthana** [comment]
I've been running into similar issues to what @fchiorascu described. For the time being I am using an unofficial mirror at https://quay.io/repository/app-sre/grafana, but I'd obviously rather be using an official one :-)

**Comment by torkelo** [comment]
That sounds terrible, was unaware the limits affected the consumers

**Comment by anishasthana** [comment]
Hi @torkelo , I see you moved the issue from "needs investigation or followup". Have there been any updates here?

**Comment by tumido** [comment]
Let me share the sentiment. :slightly_smiling_face: I'd love to see Grafana available on any other public docker image service. Quay, Grc, Ghcr... Would it be possible to push an official image to any of it as well?

The technical aspect of it should be fairly easy to implement (`docker tag` and `docker push`).

**Comment by maulikjs** [comment]
Yup! It would be fantastic to have these images in a non-rate limited environment. :) 
We run into those far too frequently.

**Comment by martavoi** [comment]
Any update on this ?

**Comment by leifmadsen** [comment]
Affects our testing as our lab systems are behind a single firewall / router, so it's not possible to pull the official Grafana images from Dockerhub. Would be great to have it sync _additionally_ to Quay.io where rate limits are not a thing currently. No suggestion here to migrate from one location to another, but rather to just push the same image to two repositories/organizations.

**Comment by onedr0p** [comment]
Github Container Registry is also an option instead of Quay or DockerHub, or it can be pushed to all 3.

**Comment by rvansa** [comment]
Hi, any reason why the images still are not on quay.io after almost a year from opening this issue?

**Comment by akhfa** [comment]
Also waiting for this

**Comment by paul-civitas** [comment]
This is pretty mission critical.

**Comment by jasperjonker** [comment]
I am also hoping quay.io would be added as a second image repo

**Comment by zqad** [comment]
Hi there,
Another reason for your consideration: Docker registry is IPv4-only: No IPv6 support!

**Comment by jcpunk** [comment]
I'd like to add my voice to having multiple sources for the container so I can do IPv6 pulls and switch to alternate sources if I hit a rate limit.

One of my student labs is behind an IPv4 NAT and when they do a whole bunch of test deploys it can cause some frustration for the other students in the lab...

**Comment by stdmje** [comment]
This is high priority IMO!

**Comment by onedr0p** [comment]
Given the recent dockerhub announcement of removing free team orgs, I'm wondering if the current `grafana` org will be affected? And if so perhaps this could finally have a high priority?

https://github.com/docker/hub-feedback/issues/2314

**Comment by samip5** [comment]
> Hi there, Another reason for your consideration: Docker registry is IPv4-only: No IPv6 support!

Actually, not true. See here: https://www.docker.com/blog/beta-ipv6-support-on-docker-hub-registry/

**Comment by jkroepke** [comment]
I would also like to see Grafana images on a different major registry.

About the rate-limit considerations: Seems like Grafana has been join the Verified Publisher program. Is this notable, since the Container Info page has a Badge "Verified Publisher".

At least, Verified Publisher are excluded from any rate-limits on DockerHub. And DockerHub has IPv6 support. At the moment, I could not see any motivation for moving into an other Docker Registry.

**Comment by quentin9696** [comment]
> At least, Verified Publisher are excluded from any rate-limits on DockerHub. And DockerHub has IPv6 support. At the moment, I could not see any motivation for moving into an other Docker Registry.

If docker is down for whatever reason, it's still good to have a fall-back somewhere else

**Comment by patchon** [comment]
I’d like to share a suggestion regarding container image distribution. 🤔 

It seems that limiting support to a single container registry might restrict some users who have specific preferences or requirements - it doesn't really matter why the user has that particular preference or requirement, the important thing is that different users will always have different needs. 🛠🧰 

Supporting at least **three** major registries could significantly enhance accessibility and flexibility for a diverse user base. Of course, I understand that extending support does introduce additional work, and it’s impractical to cover every possible option. However, offering a few more choices would help users who are currently facing limitations (be it firewalls, rate limitiing, policys or whatever). 📦 

With that being said, of course you have to "draw the line" somewhere and say "we currently support these container registries", but drawing that line at one container registry seems a _bit restrictive_ if you will. ⚠🛑 

Love Grafana and Loki btw, such great products, continue the great work! ⭐🤩 

Ps. 
I would love to try mimir as well as soon as I can use quay.io as source for the container images 🚀

**Comment by lure8225** [comment]
> I would also like to see Grafana images on a different major registry.
> 
> About the rate-limit considerations: Seems like Grafana has been join the Verified Publisher program. Is this notable, since the Container Info page has a Badge "Verified Publisher".
> 
> At least, Verified Publisher are excluded from any rate-limits on DockerHub. And DockerHub has IPv6 support. At the moment, I could not see any motivation for moving into an other Docker Registry.

The issue dos not stem from the limits on the publisher side but from the limits imposed on the consumers of the images. With pushing the official images only to docker hub and not quay or GHCR companies/students (limited IP ranges used for many pulls) who decide to not pay for dockerhub will not be able to use your images easily.

What is the overhead of adding a simple additional push to GHCR and quay? Is it so huge that a 4 year discussion needs to be had?
I could offer to work on a PR to add this but would like to know the chances of it succeeding before I put in the efforts.

**Comment by jkroepke** [comment]
> (limited IP ranges used for many pulls) who decide to not pay for dockerhub will not be able to use your images easily.

Again, Verified Publisher are excluded. If your pull rate limit is exceeded, you are still able to pull grafana images.

**Comment by onedr0p** [comment]
Wow it's been awhile since I last looked here but I still think it makes sense for the project to consider publishing images to an additional registry. While Docker Hub is widely used, it comes with some risks: their pull rate limits, account policy changes, and past decisions (like pruning inactive content or trying to lock once free features behind paid tiers) show that projects are always at the whim of Docker Inc.’s business choices. Verified Publisher exemptions help, but they still rely on Docker maintaining that program in the long term.

Sorry if this sounds like FUD but I'm very confident in Docker Inc. ability to potentially screw over people using unauthenticated pulls.

**Comment by Vahan90** [comment]
Or when docker-hub is down like right now, while we're paged on-call trying to look for alternatives.

<img width="1053" height="808" alt="Image" src="https://github.com/user-attachments/assets/28cf3eeb-ce80-494f-b046-4384e810b39b" />
